### PR TITLE
qemu.tests：add case for function test of usb device

### PR DESCRIPTION
**Pull request by maxujun**
qemu.tests：add case for function test of usb device
Deps on [631](https://github.com/avocado-framework/avocado-vt/pull/631)
Deps on [707](https://github.com/autotest/tp-qemu/pull/707l)
ID:1324316 1324315 1324309 1324320

### CODE REVIEW

**Comment by suqinhuang** [comment]
pls provide commit log

**Comment by mazhang** [comment]
Better to add error handle.

**Comment by maxujun** [comment]
@mazhang error handle include in the UsbStorageTest class

**Comment by MiriamDeng** [comment]
As it depends on pr 707 I apply it but hit some errors as following,
qemu/707.patch:138: trailing whitespace.
            usb_type = "usb-bot"    
warning: 1 line adds whitespace errors.

**Comment by maxujun** [comment]
@MiriamDeng Have updated,thanks for comments

File: qemu/tests/usbdev_func.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            dev.format_usb_disk(devname, session)
+            dev.usb_disk_io('yes', devname, session)
+
+    vm = env.get_vm(params["main_vm"])
+    dev = qemu.UsbStorageTest(test, params, env)
+    dev.add_udisk()
+    timeout = float(params.get("login_timeout", 600))
+    session = vm.wait_for_login(timeout=timeout)
+    disklist = dev.get_usb_disk_list(session)
+    function = params.get("function")
```
Comment (maxujun) [inline qemu/tests/usbdev_func.py]: have already updated,thanks.

File: qemu/tests/usbdev_func.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+from autotest.client.shared import error
```
Comment (maxujun) [inline qemu/tests/usbdev_func.py]: have already updated,thanks.

**Comment by maxujun** [comment]
@xutian Could you help review this patch?

File: qemu/tests/usbdev_func.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    vm = env.get_vm(params["main_vm"])
+    dev = qemu.UsbStorageTest(test, params, env)
+    dev.add_udisk()
+    timeout = float(params.get("login_timeout", 600))
+    session = vm.wait_for_login(timeout=timeout)
+    disklist = dev.get_usb_disk_list(session)
+    function = params.get("function")
+    if function and hasattr(function, '__call__'):
+        eval(function)(disklist, session)
```
Comment (xutian) [inline qemu/tests/usbdev_func.py]: avoid use eval in your test script.  you can instead it like
```
func = getattr(module_name, function_name)
if callable(func):
     func(disklist, session)
```

**Comment by CongLi** [comment]
Close it as @maxujun has no time to handle it.

**Pull request closed without merging**